provider/pkg/clients/clients.go

```diff
@@ -93,12 +94,7 @@ func (dcs *DynamicClientSet) ResourceClient(gvk schema.GroupVersionKind, namespa
 	}
 
 	// For namespaced Kinds, create a namespaced client. If no namespace is provided, use the "default" namespace.
-	namespaced, err := IsNamespacedKind(gvk, dcs)
-	if err != nil {
-		return nil, err
-	}
-
-	if namespaced {
```

Comment:
The mapper returns the requisite scope information, the call to `IsNamespaceKind` is redundant.